### PR TITLE
Feature: Jobs form

### PR DESCRIPTION
**Pull request by natmegs**
*Note*: This is quite a large PR, but keep in mind all code in this branch has already been reviewed before merging into the feature branch.

Feature branch for the new Jobs form [DCOS-48308](https://jira.mesosphere.com/browse/DCOS-48308). This feature creates a new full screen modal and jobs form for creating and updating jobs. The existing functionality of the jobs form will be preserved along with the following features:
- Container options: user can choose between UCR and Docker, and select advanced options corresponding to each of those
- GPUs: user can add GPU resources to a job if using UCR
- Run configuration: advanced job run settings such as restart configuration
- Upgrade to metronome v1 spec from v0

## Dependencies

Tickets that need to be merged into this branch before it can be merged to master:
- [x] https://jira.mesosphere.com/browse/DCOS-48316
- [x] https://jira.mesosphere.com/browse/DCOS-48445
- [x] https://jira.mesosphere.com/browse/DCOS-48352
- [x] https://jira.mesosphere.com/browse/DCOS-48340
- [x] https://jira.mesosphere.com/browse/DCOS-48322
- [x] https://jira.mesosphere.com/browse/DCOS-48333

### CODE REVIEW

File: locale/zh/messages.json [hunk 1]
```diff
@@ -286,13 +300,15 @@
...
   "Docker Runtime only supports the default size for implicit volumes, please select Universal Container Runtime (UCR) if you want to modify the size.": "Docker 运行时仅支持隐式卷的默认大小，如果要修改大小，请选择通用容器运行时 (UCR)。",
+  "Docker parameters": "",
+  "Docker's container runtime. No support for GPU resources and requires image.": "",
   "Docker’s container runtime. No support for multiple containers (Pods) or GPU resources.": "Docker 的容器运行时。不支持多个容器（Pod）或 GPU 资源。",
   "Documentation": "文档",
   "Don't run app tasks on a particular set of attribute IDs": "不要在特定的属性 ID 集合上运行应用程序任务",
   "Download": "下载",
   "Download Config": "下载配置",
   "Download Snapshot": "下载快照",
-  "Download and install: <0><1/> Download dcos.exe</0>.": "下载和安装：<0><1/> 下载 dcos.exe</0>.",
```
Comment (brandonc) [inline locale/zh/messages.json]: Deleting this space destroyed the translation 😭 but you could manually put it back

File: locale/zh/messages.json [hunk 2]
```diff
@@ -302,7 +318,7 @@
   "Edit Configuration": "编辑配置",
   "Edit Directory": "编辑目录",
   "Edit External Load Balancer": "编辑外部负载均衡器",
-  "Edit Job ({0})": "编辑作业（{0}）",
```
Comment (brandonc) [inline locale/zh/messages.json]: ditto

File: locale/zh/messages.json [hunk 3]
```diff
@@ -350,7 +370,7 @@
   "FAILURES": "失败",
   "FALSE": "错误",
   "Failed": "已失败",
-  "Failure %": "失败 %",
```
Comment (brandonc) [inline locale/zh/messages.json]: Why do these get destroyed? Any insight?

File: locale/zh/messages.json [hunk 4]
```diff
@@ -630,12 +672,14 @@
   "Overview": "概述",
   "Owner": "所有者",
   "P99 LATENCY": "P99 延迟",
-  "P99 Latency": "P99 延迟",
```
Comment (brandonc) [inline locale/zh/messages.json]: ☝️

**Comment by natmegs** [comment]
@brandonc That's so bizarre... I'm not sure why those would have been removed but I added them back in manually. Also strange that there seem to be instances of multiple identical messages (`Failure %`, `P99 Latency`)...

**Pull request merged**